### PR TITLE
[UDF](java udf)  using config to enable java udf instead of macro at compile time

### PR DESCRIPTION
**Pull request by yangzhg**
# Proposed changes

using config to enable java udf instead of macro at compile time

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [x] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.18 seconds
 load time: 447 seconds
 storage size: 17181072247 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221108071128_clickbench_pr_41778.html

File: be/src/exec/data_sink.cpp [hunk 1]
```diff
@@ -171,11 +171,13 @@ Status DataSink::create_data_sink(ObjectPool* pool, const TDataSink& thrift_sink
...
-#ifdef LIBJVM
-            sink->reset(new vectorized::VJdbcTableSink(pool, row_desc, output_exprs));
-#else
-            return Status::InternalError("Jdbc table sink is disabled since no libjvm is found!");
-#endif
+            if (config::enable_java_support) {
+                sink->reset(new vectorized::VJdbcTableSink(pool, row_desc, output_exprs));
+            } else {
+                return Status::InternalError(
+                        "Jdbc table sink is one enabled, you can change be config "
```
Comment (cambyzju) [inline be/src/exec/data_sink.cpp:178]:
```suggestion
                        "Jdbc table sink is not enabled, you can change be config "
```

Comment (cambyzju) [reply be/src/exec/data_sink.cpp:178]: or

`is disabled` ?

**Review by cambyzju — APPROVED** [review]
LGTM

**Pull request merged**